File: Add Code Here/C++/tree/selectionSort.py
# Sorting Algorithm: Selection Sort
# Time Complexity: O(n^2) , where n is the no of elements
# Space Complexity: O(1)
def selectionSort(arr):
    print(f"Unsorted Array: {arr}")
    # Size of Array
    size = len(arr)
    swaps = 0

    i = 0
    while i < size-1:
        j = i + 1
        while j <= size-1:
            if arr[i] > arr[j]:
                # swap
                arr[i], arr[j] = arr[j], arr[i]
                swaps = swaps + 1

            j = j + 1
        i = i + 1
        print(f"{arr}")
    print(f"Steps: {i} \t Swaps: {swaps}")
    return arr

a = [2,6,9,4,8,1,5,7]
sorted_arr = selectionSort(a)
print(f"Sorted Array is: {sorted_arr}")

File: Add Code Here/C++/tree/Selection_sort.c
#include <stdio.h>
void selection(int arr[], int n){
	int i, j, small;
	for (i = 0; i < n - 1; i++){
		small = i;
		for (j = i + 1; j < n; j++)
			if (arr[j] < arr[small])
				small = j;
		int temp = arr[small];
		arr[small] = arr[i];
		arr[i] = temp;
	}
}
void printArr(int a[], int n){
	int i;
	for (i = 0; i < n; i++)
		printf("%d ", a[i]);
}
int main(){
	int a[] = { 12, 31, 25, 8, 32, 17 };
	int n = sizeof(a) / sizeof(a[0]);

	printf("Before sorting array elements are - \n");
	printArr(a, n);

	selection(a, n);

	printf("\nAfter sorting array elements are - \n");
	printArr(a, n);

	return 0;
}
